src/braket/circuits/circuit.py

```diff
@@ -912,6 +901,20 @@ def as_unitary(self) -> np.ndarray:
 
         return calculate_unitary(qubit_count, self.instructions)
 
+    @property
+    def has_noncommuting_observables(self) -> bool:
```

Comment:
I do wonder about this name, since we actually only support identical observables, not commuting observables. At the very least this should be renamed to non_qubitwise_commuting, but that's not completely accurate either. Probably need a second opinion from someone else.

Reply:
How about I flip the truth value and call it `observables_simultaneously_measurable`?

Reply:
Oh, that could work. Thanks!